model/vsstree.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+
+#
+#
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+
+#
+# Convert vspec file to GraphQL
+#
+import re
+import stringcase
+from anytree import (Node, Resolver, ChildResolverError)
+
+from model.constants import VSSType, VSSDataType, StringStyle
+
+
+class VSSNode(Node):
+    description = None
+    uuid = None
+    type: VSSType
+    data_type = VSSDataType.NONE
+
+    unit = None
+    min = None
+    max = None
+    enum = None
+    deprecation = None
+
+    aggregate = False
+    child_type = None
+    child_properties = 0
+    prop_name = None
+    prop_type = None
+    prop_format = None
+    prop_unit = None
+    prop_value = None
+
+    default_value = None
+    value = None
+
+    instances = None
+
+    def __init__(self, name, vss_type: VSSType, vss_uuid, vss_description, parent=None, children=None):
+        super().__init__(name, parent, children)
+        self.description = vss_description
+        self.type = vss_type
+        self.uuid = vss_uuid
+
+    def is_private(self):
+        node = self
+        while node:
+            if node.name == "Private":
+                return True
+            node = node.parent
+        return False
+
+    @staticmethod
+    def case_conversion(source, style: StringStyle):
```

Comment:
This is an optional suggestion. I had the need to generate types out of VSS in multiple programming and definition languages. All had different conventions. In order to support that, I added StringStyle and conversion libraries that should cover all kinds of variable naming conventions. However, we can get rid of that and just allow the generator code to decide for that. It probably boils down to adding some convenience to the core framework so that generators do not need to re-invent the wheel here.